vscode-extension/src/msildecompiler/decompiledTreeProvider.ts

```diff
@@ -192,15 +192,15 @@ export class DecompiledTreeProvider implements TreeDataProvider<MemberNode>, Tex
     getTypes(assembly: string, namespace: string): Thenable<MemberNode[]> {
         let request: ListTypesRequest = { "AssemblyPath": assembly, "Namespace": namespace };
         return serverUtils.getTypes(this.server, request).then(result => {
-            return result.Types.map(t => new MemberNode(assembly, t.Name, t.Token.RID, t.Token.TokenType, t.MemberSubKind, -1));
+            return result.Types.map(t => new MemberNode(assembly, t.Name, t.Token & 0x00FFFFFF, (t.Token >> 24) & 0xFF, t.MemberSubKind, -1));
```

Comment:
Please extract these bit operations into methods with comments to help others understand why.